src/views/Profile/components/EditProfileModal/ChangeProfilePicPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+import { Button, InjectedModalProps, Skeleton, Text } from '@pancakeswap-libs/uikit'
+import { useWallet } from '@binance-chain/bsc-use-wallet'
+import { useDispatch } from 'react-redux'
+import nftList from 'config/constants/nfts'
+import { useProfile, useToast } from 'state/hooks'
+import useI18n from 'hooks/useI18n'
+import { fetchProfile } from 'state/profile'
+import useGetWalletNfts from 'hooks/useGetWalletNfts'
+import useApproveConfirmTransaction from 'hooks/useApproveConfirmTransaction'
+import { usePancakeRabbits, useProfile as useProfileContract } from 'hooks/useContract'
+import { getPancakeProfileAddress, getPancakeRabbitsAddress } from 'utils/addressHelpers'
+import SelectionCard from '../SelectionCard'
+import ApproveConfirmButtons from '../ApproveConfirmButtons'
+
+type ChangeProfilePicPageProps = InjectedModalProps
+
+const ChangeProfilePicPage: React.FC<ChangeProfilePicPageProps> = ({ onDismiss }) => {
+  const [tokenId, setTokenId] = useState(null)
+  const TranslateString = useI18n()
+  const { isLoading, nfts: nftsInWallet } = useGetWalletNfts()
+  const dispatch = useDispatch()
+  const { profile } = useProfile()
+  const pancakeRabbitsContract = usePancakeRabbits()
+  const profileContract = useProfileContract()
+  const { account } = useWallet()
+  const { toastSuccess } = useToast()
+  const {
+    isApproving,
+    isApproved,
+    isConfirmed,
+    isConfirming,
+    handleApprove,
+    handleConfirm,
+  } = useApproveConfirmTransaction({
+    onApprove: () => {
+      return pancakeRabbitsContract.methods.approve(getPancakeProfileAddress(), tokenId).send({ from: account })
```

Comment:
did you try this ? approve takes a value as the second params, why is there a token id ?

Reply:
This approval is not for cake but for the transfer of NFT.

Reply:
https://bscscan.com/address/0xDf7952B35f24aCF7fC0487D01c8d5690a60DBa07#writeContract

![image](https://user-images.githubusercontent.com/72719739/107305822-07649300-6ac7-11eb-8e1e-3dcaa27cbf43.png)